core/src/test/kotlin/com/malinskiy/marathon/execution/strategy/impl/flakiness/ProbabilityBasedFlakinessStrategySpek.kt

```diff
@@ -0,0 +1,43 @@
+package com.malinskiy.marathon.execution.strategy.impl.flakiness
+
+import com.malinskiy.marathon.MetricsProviderStub
+import com.malinskiy.marathon.TestGenerator
+import com.malinskiy.marathon.execution.TestShard
+import org.amshove.kluent.shouldBe
+import org.jetbrains.spek.api.Spek
+import org.jetbrains.spek.api.dsl.describe
+import org.jetbrains.spek.api.dsl.it
+import java.time.Instant
+
+class ProbabilityBasedFlakinessStrategySpek : Spek({
+    describe("probability-based-strategy test") {
+        val instant = Instant.now()
+        group("min success rate 0.8") {
+            it("should return 2 flaky tests for one with success rate = 0.5") {
+                val metricsProvider = MetricsProviderStub()
+                val strategy = ProbabilityBasedFlakinessStrategy(0.8, instant)
+                val testShard = TestShard(TestGenerator().create(1))
+                val result = strategy.process(testShard, metricsProvider)
+                result.tests.size shouldBe 1
+                result.flakyTests.size shouldBe 2
```

Comment:
These strategies needs to be commented, i spent too much time trying to figure out why single test should produce two flakyTests 🦀